### PR TITLE
ci/validate: New file to check syntax

### PR DESCRIPTION
**Pull request by cgwalters**
I've had syntax errors in shell script in `overlay.d` too many
times, let's add a quick check for this.  This is a more lightweight
version of what we're doing in coreos-assembler.

### CODE REVIEW

**Comment by cgwalters** [comment]
OK, this is passing CI now.

File: ci/validate [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+for root, dirs, files, rootfd in os.fwalk('.'):
+    # Skip .git
+    if '.git' in dirs:
+        dirs.remove('.git')
+    for name in files:
+        if name.endswith(('.yaml', '.yml')):
+            with open(os.open(name, dir_fd=rootfd, flags=os.O_RDONLY)) as f:
+                yaml.safe_load(f)
+                validated +=1
```
Comment (jlebon) [inline ci/validate:24]: Minor: we could just `continue` here, right?

**Review by jlebon — APPROVED** [review]
One minor comment, LGTM as is.

**Comment by jlebon** [comment]
```
[fedora-coreos-config] Running shell script
+ set -xeuo pipefail
+ ./ci/validate
Validated 26 files
```

Cool, thanks for adding this check!

**Pull request merged**